### ISSUE TITLE
Warning: abi keys are missing values

### ISSUE THREAD

**Issue by degski** [microsoft/vcpkg#7865]
Lately I'm getting this warning, underneath a number of libs are listed. What does this mean and is it a problem?

**Comment by degski** [comment]
Today:

    Warning: abi keys are missing values:
        boost-chrono
        boost-context
        boost-conversion
        boost-crc
        boost-date-time
        boost-filesystem
        boost-multi-index
        boost-program-options
        boost-regex
        boost-smart-ptr
        boost-system
        boost-thread
        gflags
        glog
        libevent
        openssl
        zlib

**Comment by Vultraz** [comment]
Also getting them, when building x64 harfbuzz:
```
Warning: abi keys are missing values:
    freetype
    glib
```

**Comment by degski** [comment]
@Vultraz Nothing seems to break down, just this message.

**Comment by cbezault** [comment]
It's the result of a change I made where we changed the format of the metadata for installed packages. I'm planning on improving the error message and writing some documentation. It currently has no negative effect as it's going to be used in future features. If you want the error to go away you will need to rebuild all your packages.

**Comment by degski** [comment]
@cbezault It's a warning, so I'm not laying awake of it. And from what I understand from you, these warnings will go away over time, as and when packages get updated and rebuilt. Is that correct?

**Comment by cbezault** [comment]
Yes it's a warning, not an error. They will go away over time due to upgrades, though I can imagine a scenario where they never do. Essentially we're calculating a hash of the package and one of the things that feeds into the hash is the hash of all the dependencies. We don't calculate the hash (and you see this warning) if any of the hashes of the dependencies are unavailable.

**Comment by Zingam** [comment]
> It's the result of a change I made where we changed the format of the metadata for installed packages. I'm planning on improving the error message and writing some documentation. It currently has no negative effect as it's going to be used in future features. If you want the error to go away you will need to rebuild all your packages.

Is this `vcpkg upgrade --no-dry-run` supposed to fix the warnings?
